src/WebJobs.Extensions.NotificationHubs/Config/NotificationHubsConfiguration.cs

```diff
@@ -47,19 +59,36 @@ public void Initialize(ExtensionConfigContext context)
             converterManager.AddNotificationHubConverters();
 
             var bindingFactory = new BindingFactory(nameResolver, converterManager);
+            IBindingProvider clientProvider = bindingFactory.BindToExactType<NotificationHubAttribute, NotificationHubClient>(BindForNotificationHubClient);
+
             var ruleOutput = bindingFactory.BindToAsyncCollector<NotificationHubAttribute, Notification>(BuildFromAttribute);
-            extensions.RegisterBindingRules<NotificationHubAttribute>(ruleOutput);
+
+            extensions.RegisterBindingRules<NotificationHubAttribute>(ruleOutput, clientProvider);
         }
 
-        private IAsyncCollector<Notification> BuildFromAttribute(NotificationHubAttribute attribute)
+        internal IAsyncCollector<Notification> BuildFromAttribute(NotificationHubAttribute attribute)
         {
             string resolvedConnectionString = ResolveConnectionString(attribute.ConnectionStringSetting);
             string resolvedHubName = ResolveHubName(attribute.HubName);
 
-            INotificationHubClientService service = new NotificationHubClientService(resolvedConnectionString, resolvedHubName);
+            INotificationHubClientService service = GetService(resolvedConnectionString, resolvedHubName);
             return new NotificationHubAsyncCollector(service, attribute.TagExpression);
         }
 
+        internal NotificationHubClient BindForNotificationHubClient(NotificationHubAttribute attribute)
+        {
+            string resolvedConnectionString = ResolveConnectionString(attribute.ConnectionStringSetting);
+            string resolvedHubName = ResolveHubName(attribute.HubName);
+            INotificationHubClientService service = GetService(resolvedConnectionString, resolvedHubName);
+
+            return service.GetNotificationHubClient();
+        }
+
+        internal INotificationHubClientService GetService(string connectionString, string hubName)
+        {
+            return ClientCache.GetOrAdd(new Tuple<string, string>(connectionString, hubName), (c) => NotificationHubClientServiceFactory.CreateService(c.Item1, c.Item2));
```

Comment:
>new Tuple<string, string>(connectionString, hubName) [](start = 40, length = 52)

Should the hubname comparison be case sensitive or case insensitive? Not sure how it works. But you should add a test using a hubname of "TEST" and "test" and see if it behaves the way you want.